notebooks/32-builtin-types-nb.md

```diff
@@ -645,6 +645,32 @@ for (let key in bond) {
 }
 ```
 
+````{admonition} practice
+:class: seealso
+
+Remember that it is not obvious when to use `in` or `of`. Try to guess what the following code returns:
+````
+
+```{code-cell}
+tab = ["A", "B", "C"]
+obj = {x: "A", y: "B", z: "C"}
+tabAsObj = {"1": "A", "2": "B", "3": "C"}
+
+for (let i in tab) 
+    console.log(`value=${i} type=${typeof i}`)
+
+for (let i of tab)
+    console.log(`value=${i} type=${typeof i}`)
```

Comment:
it is tempting to write a function instead here - especially as we instruct students to try it out with other inputs 
is it something you considered and rejected on purpose ? I mean, to let them feel the pain of not using functions ?

Reply:
Nope, just to avoid "obfuscating" - and because I propose to to in the browser directly, declaring function is sometime taking too much time especialy when you don't get it right on the first time.

Reply:
But I did take you remark into my new version ;).